### DISCUSSION TITLE
How to get command result in daemon socket mode

### DISCUSSION THREAD

**Discussion by Tsjippy** [AsamK/signal-cli#1527] [Q&A] [answered]
if I do `echo '{"jsonrpc":"2.0","method":"listGroups","id":"my special mark"}' |  signal-cli jsonRpc` in Ubuntu terminal I get a response just fine.

My listener also works just fine in php
```
<?php 
(...)
while(1){
    $request    = fread($socket, 4096);
    flush();
    (..)
}
```

But when I do `fwrite($socket, json_encode($data));`  the command is executed, but I never get the response in my receiver. The listner only sees incoming messages, not the response to requests.

What am I doing wrong?

The same is the case with python3:
```
socket_path = '/home/simnige1/sockets/signal'
client = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
client.connect(socket_path)
message = '{"jsonrpc":"2.0","method":"listGroups","id":"5"}'
client.sendall(message.encode())
response = client.recv(1024) # nothing there

# I see all incoming messages, but no responses
while True:
 response = client.recv(1024)
 print(f'Received response: {response.decode()}')

```

**Comment by AsamK** [comment, marked as answer]
Json rpc commands need to be terminated by a newline. So I think you're missing the \n at the end.

**Reply by Tsjippy** [reply]
Yes you are right! Adding "\n" fixed the issue. Thanks
